airflow/dags/gtfs_loader/gtfs_schedule_history_load.py

```diff
@@ -45,57 +40,88 @@ def main(execution_date, ti, **kwargs):
         for table_file, is_required, colnames in table_details:
-            # validation report handled above, since it is in a subfolder.
-            # this is a side-effect of it technically not being an extraction,
-            # and we might want to re-create it (and put results in "processed")
-            if table_file == VALIDATION_REPORT:
+            # validation report handled in a separate task, since it is in a subfolder
+            # and should be ran separately in case the feed is unparseable.
+            if table_file == constants.VALIDATION_REPORT:
                 continue
 
-            src_path = f"{src_dir}/{table_file}"
-            dst_path = f"{dst_dir}/{table_file}"
+            src_path = "/".join(
+                ["schedule", str(execution_date), id_and_url, table_file]
+            )
+            dst_path = "/".join(
+                ["schedule", "processed", f"{date_string}_{id_and_url}", table_file]
+            )
 
             print(f"Copying from {src_path} to {dst_path}")
 
             if not is_required and not fs.exists(f"{bucket}/{src_path}"):
                 print(f"Skipping missing optional file: {src_path}")
             else:
-                _keep_columns(
+                parse_error_encountered = _keep_columns(
                     src_path,
                     dst_path,
                     colnames,
                     row["itp_id"],
                     row["url_number"],
                     date_string,
                 )
+                if parse_error_encountered:
+                    print(
+                        f"Fatal parsing error encountered in {table_file} for id and "
+                        "URL: {id_and_url}."
+                    )
+                    parse_error_encountered_in_this_feed = True
+
+                feed_tables_process_results.append(
+                    {
+                        "calitp_itp_id": row["itp_id"],
+                        "calitp_url_number": row["url_number"],
+                        "calitp_extracted_at": execution_date.to_date_string(),
+                        "filename": table_file,
+                        "parse_error_encountered": parse_error_encountered,
+                    }
+                )
+
+        # note the parse result for this feed
+        feed_process_resuls.append(
+            {
+                "calitp_itp_id": row["itp_id"],
+                "calitp_url_number": row["url_number"],
+                "calitp_extracted_at": execution_date.to_date_string(),
+                "parse_error_encountered": parse_error_encountered_in_this_feed,
+            }
+        )
 
         ttl_feeds_copied += 1
 
         print("total feeds copied:", ttl_feeds_copied)
+
+    # save feed and feed table process results
+    save_to_gcfs(
+        pd.DataFrame(feed_process_resuls).to_csv(index=False).encode(),
+        f"schedule/{execution_date}/processed/feed_parse_result.csv",
+        use_pipe=True,
+    )
+    save_to_gcfs(
```

Comment:
Can we load this data as an external table? Should only require adding this to the gtfs_schedule_history DAG:

```yaml
operator: operators.ExternalTable
source_objects:
  - 'schedule/*/processed/feed_parse_result.csv'
destination_project_dataset_table: "gtfs_schedule_history.calitp_feed_parse_result"
skip_leading_rows: 1
schema_fields:
  - name: calitp_itp_id
    type: INTEGER
  - name: calitp_url_number
    type: INTEGER
  - name: calitp_extracted_at
    type: DATE
  - name: parse_error_encountered
    type: BOOLEAN
```

(We can definitely just chuck this into a issue if you want to merge)

Reply:
Done in 4ae1221